php-classes/Slate/CBL/Tasks/Task.php

```diff
@@ -195,4 +198,16 @@ public static function delete($id)
 
         return DB::affectedRows() > 0;
     }
+
+    protected static function validateParentTask($validator, Task $Task, $options)
+    {
+        if (!$validator->hasErrors('ParentTaskID') && $Task->ParentTaskID) {
+            if ($Task->ParentTaskID == $Task->ID) {
+                $validator->addError('ParentTaskID', 'A task cannot be its own parent');
+
+                // clear immediately to prevent validation loop
+                $Task->ParentTaskID = null;
```

Comment:
@nbey FYI, I found this necessary when I tested the server-side rejection -- the response would hang indefinitely as validation went down an infinite recursion.

I tried for a while to update the JSON encoder or record validator to detect/avoid recursion, but no luck. This feels a bit hacky but stops validation from looping